packages/babel-plugin-transform-jsx-to-htm/index.mjs

```diff
@@ -86,50 +86,24 @@ export default function jsxToHtmBabelPlugin({ types: t }, options = {}) {
 			cooked: buffer
 		}));
 		buffer = '';
-	}
-
-	function processNode(node, path, isRoot) {
-		const open = node.openingElement;
-		const { name } = open.name;
-
-		if (name.match(/^[A-Z]/)) {
-			raw('<');
-			expr(t.identifier(name));
-		}
-		else {
-			raw('<');
-			raw(name);
-		}
-
-		if (open.attributes) {
-			for (let i = 0; i < open.attributes.length; i++) {
-				const attr = open.attributes[i];
-				raw(' ');
-				if (t.isJSXSpreadAttribute(attr)) {
-					raw('...');
-					expr(attr.argument);
-					continue;
-				}
-				const { name, value } = attr;
-				raw(name.name);
-				if (value) {
-					raw('=');
-					if (value.expression) {
-						expr(value.expression);
-					}
-					else if (t.isStringLiteral(value)) {
-						escapePropValue(value);
-					}
-					else {
-						expr(value);
-					}
-				}
-			}
-		}
-
-		const children = t.react.buildChildren(node);
+  }
+  
+  function getName(node) {
+    switch (node.type) {
+      case 'JSXMemberExpression':
+        return `${node.object.name}.${node.property.name}`
+    
+      default:
+        return node.name;
+    }
+  }
```

Comment:
This actually brings to focus a larger missing thing in the old code: Namespaced component names like `<A.B.C>` don't work. So how about extending on the idea of `getName` with the following helpers:

```js
const FRAGMENT_EXPR = dottedIdentifier('React.Fragment');

function isFragmentName(node) {
	return t.isNodesEquivalent(FRAGMENT_EXPR, name);
}

function isComponentName(node) {
	return !t.isIdentifier(node) || node.name.match(/^[A-Z]/);
}

function getNameExpr(node) {
	if (!t.isJSXMemberExpression(node)) {
		return t.identifier(node.name);
	}
	return t.memberExpression(
		getNameExpr(node.object),
		t.identifier(node.property.name)
	);
}
```

They could be used later like so:

```js
const open = node.openingElement;
const name = getNameExpr(open.name);

if (!isFragmentName(name)) {
	if (isComponentName(name)) {
		raw('<');
		expr(name);
	}
	else {
		raw('<');
		raw(name.name);
	}
}
```

This would allow both identifying `<React.Fragment>` elements *and* also adding nice handling for elements like `<A.B.C>`.

Reply:
In fact changing the `/^[A-Z]/` test to `/^[^a-z]/` would also fix #92, I think? /cc @developit

Reply:
> This actually brings to focus a larger missing thing in the old code: Namespaced component names like <A.B.C> don't work

Its something i noticed. But i believe that should be done on another PR to keep this focused

Reply:
Good point. I can create a PR once this one has been merged 👍

Reply:
Babel seems to parse tag names matching `/(^[A-Z_$]|[A-z_$]\..)/` into references.
[repl example](https://babeljs.io/repl#?babili=false&browsers=%3E%202%25&build=&builtIns=false&spec=false&loose=true&code_lz=PQKhAIAECsGcA9wAtwmAKHaE7XgKYA2-AtvgHYAusAXLmugDwBmA9q-MAHwDcTbrACSdembPXAAnfM3zTyAY3y16GRgDF2Ivo2HcdAfW1NBA44wNn9_dgDoAhucDdwMaA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=react%2Cenv&prettier=false&targets=&version=7.4.5&externalPlugins=)